shared_model/backend/protobuf/block.hpp

```diff
@@ -127,11 +119,12 @@ namespace shared_model {
       }};
 
       const Lazy<SignatureSetType<proto::Signature>> signatures_{[this] {
-        SignatureSetType<proto::Signature> sigs;
-        for (const auto &sig : proto_->signatures()) {
-          sigs.emplace(sig);
-        }
-        return sigs;
+        auto signatures = proto_->signatures()
+            | boost::adaptors::transformed([](const auto &x) {
+                            return proto::Signature(x);
+                          });
+        return SignatureSetType<proto::Signature>(signatures.begin(),
```

Comment:
Why not just return signatures?

Reply:
`signatures` type is `boost::range`, and `std::unordered_set` cannot be constructed from it directly.